src/query/storage/converter_test.go

```diff
@@ -353,28 +361,106 @@ func TestPromTimeSeriesToSeriesAttributesM3Type(t *testing.T) {
 	}
 
 	for proto, expected := range mapping {
-		attrs, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{M3Type: proto})
+		attrs, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{
+			Source: prompb.Source_PROMETHEUS,
+			M3Type: proto,
+		})
 		require.NoError(t, err)
 		assert.Equal(t, expected, attrs.M3Type)
+		assert.Equal(t, ts.SourceTypePrometheus, attrs.Source)
+	}
+
+	_, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{
+		Source: prompb.Source_PROMETHEUS,
+		M3Type: -1,
+	})
+	require.Error(t, err)
+}
+
+func TestPromTimeSeriesToSeriesAttributesMetricsTypeFromOpenMetrics(t *testing.T) {
+	mapping := map[prompbMetricTypeWithNameSuffix]promMetricTypeWithBool{
+		{metricType: prompb.MetricType_UNKNOWN}: {metricType: ts.PromMetricTypeUnknown},
+
+		{prompb.MetricType_COUNTER, "total"}:   {metricType: ts.PromMetricTypeCounter, handleValueResets: true},
+		{prompb.MetricType_COUNTER, "created"}: {metricType: ts.PromMetricTypeCounter},
+
+		{metricType: prompb.MetricType_GAUGE}:    {metricType: ts.PromMetricTypeGauge},
+		{metricType: prompb.MetricType_INFO}:     {metricType: ts.PromMetricTypeInfo},
+		{metricType: prompb.MetricType_STATESET}: {metricType: ts.PromMetricTypeStateSet},
+
+		{prompb.MetricType_HISTOGRAM, "bucket"}:  {metricType: ts.PromMetricTypeHistogram, handleValueResets: true},
+		{prompb.MetricType_HISTOGRAM, "count"}:   {metricType: ts.PromMetricTypeHistogram, handleValueResets: true},
+		{prompb.MetricType_HISTOGRAM, "sum"}:     {metricType: ts.PromMetricTypeHistogram, handleValueResets: true},
+		{prompb.MetricType_HISTOGRAM, "created"}: {metricType: ts.PromMetricTypeHistogram},
+
+		{prompb.MetricType_GAUGE_HISTOGRAM, "bucket"}: {metricType: ts.PromMetricTypeGaugeHistogram},
+		{prompb.MetricType_GAUGE_HISTOGRAM, "gcount"}: {metricType: ts.PromMetricTypeGaugeHistogram, handleValueResets: true},
+		{prompb.MetricType_GAUGE_HISTOGRAM, "gsum"}:   {metricType: ts.PromMetricTypeGaugeHistogram},
+
+		{prompb.MetricType_SUMMARY, "count"}:   {metricType: ts.PromMetricTypeSummary, handleValueResets: true},
+		{prompb.MetricType_SUMMARY, "sum"}:     {metricType: ts.PromMetricTypeSummary, handleValueResets: true},
```

Comment:
Add a case with no suffix

```suggestion
                {prompb.MetricType_SUMMARY}:            {metricType: ts.PromMetricTypeSummary},
		{prompb.MetricType_SUMMARY, "count"}:   {metricType: ts.PromMetricTypeSummary, handleValueResets: true},
		{prompb.MetricType_SUMMARY, "sum"}:     {metricType: ts.PromMetricTypeSummary, handleValueResets: true},
```